### DISCUSSION TITLE
Fetch data from a static site.

### DISCUSSION THREAD

**Discussion by HenrijsS** [nuxt/nuxt#9637] [Questions (Nuxt 2)] [answered]
I have a static site and want to always fetch the most up-to-date data which doesn't need SEO.
asyncData and fetch() always cache upon page generation.

Is there a way to have a component or page fetch data that is only called upon pageload?

This doesn't even interfere with the JAMstack structure. 

The data gets fetched -> Response is put into Vuex store -> Everything works.

The docs show no mention of this. Has anyone even used this kind of feature?

Example: My page has a daily updating price-list for products. I always want my clients to see the latest data but I don't want to trigger an additional build. When the client navigates to my site, a GET request is called from THEIR BROWSER to my own server and doesn't even require any processing from my Static site hosting provider. The content then gets cached for the client's session and they can freely navigate the rest of the already generated site.

**Comment by TheAlexLichter** [comment, marked as answer]
You can use Vue's lifecycle hooks (e.g. `mounted` or `beforeMount`) to fetch data only on client-side. This won't trigger a rebuild and the user's browser will fetch the data.
Alternatively you can use Nuxt's `fetch` with `fetchOnServer: false` for a specific component. Then, the `fetch` function of this component will be called on client-side only as well.

**Reply by HenrijsS** [reply]
I'll try mounted on my next project. Thanks!

Regarding the fetch, [here's a quote from the docs](https://nuxtjs.org/docs/2.x/features/data-fetching#the-fetch-hook):

> For static hosting, the fetch hook is only called during page generation, and the result is then cached for use on the client. To avoid cache conflicts, it may be necessary to specify a name for your component, or alternatively provide a unique fetchKey implementation.

**Reply by TheAlexLichter** [reply]
The docs are correct so far given the default options.

But the mentioned approach is explained [a bit further down the site](https://nuxtjs.org/docs/2.x/features/data-fetching#options)

> When `fetchOnServer` is falsy [...], `fetch` will be called only on client-side and `$fetchState.pending` will return `true` when server-rendering the component.
